### PR TITLE
Adding support to auto complete property names and values

### LINKED ISSUE #41: Add intellisense when editing .editorconfig file
I think it would be a nice addition if the editor showed suggestions while editing a `.editorconfig` file. It's just something that came to my mind of `nice to have`.

### PR DESCRIPTION
**Pull request by dinhani**
Fixes #41 

This PR adds supports to autocomplete property names and values when working in the `.editorconfig` file.

### CODE REVIEW

File: src/EditorConfigCompletionProvider.ts [hunk 1]
```diff
@@ -4,14 +4,14 @@ import { TextDocument, Range, Position } from 'vscode'
...
-		new Property("root", ["true", "false"]),
-		new Property("charset", ["utf-8", "utf-8-bom", "utf-16be", "utf-16le", "latin1"]),
-		new Property("end_of_line", ["lf", "cr", "crlf"]),
-		new Property("indent_style", ["tab", "space"]),
-		new Property("indent_size", ["1", "2", "3", "4", "5", "6", "7", "8"]),
-		new Property("insert_final_newline", ["true", "false"]),
-		new Property("tab_width", ["1", "2", "3", "4", "5", "6", "7", "8"]),
-		new Property("trim_trailing_whitespace", ["true", "false"])
+		new Property("root", ["true", "false"], "Special property that should be specified at the top of the file outside of any sections. Set to true to stop .editorconfig files search on current file."),
+		new Property("charset", ["utf-8", "utf-8-bom", "utf-16be", "utf-16le", "latin1"], "Set to latin1, utf-8, utf-8-bom, utf-16be or utf-16le to control the character set. Use of utf-8-bom is discouraged."),
```
Comment (jedmao) [inline src/EditorConfigCompletionProvider.ts]: Why are we discouraging `utf-8-bom`?

**Comment by dinhani** [comment]
Should "unset" be displayed when showing the possible values for a property?

**Comment by jedmao** [comment]
@dinhani yes, I think that would be very helpful!

**Comment by jedmao** [comment]
Any feedback on this question? https://github.com/editorconfig/editorconfig-vscode/pull/99#discussion_r103983300

**Comment by jedmao** [comment]
Related: https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties#current-universal-properties

File: src/EditorConfigCompletionProvider.ts (hunk 1 above)
Comment (dinhani) [inline src/EditorConfigCompletionProvider.ts]: I got this info from the website, but I don't know why.

 http://editorconfig.org/

utf-8-bom

Comment (jedmao) [inline src/EditorConfigCompletionProvider.ts]: Fair enough. I'm not sure why it's there either, but it's there! So you made a good decision 😄

File: src/EditorConfigCompletionProvider.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+		new Property('tab_width', ['1', '2', '3', '4', '5', '6', '7', '8'], 'A whole number defining the number of columns used to represent a tab character. This defaults to the value of indent_size and doesn\'t usually need to be specified.'),
+		new Property('trim_trailing_whitespace', ['true', 'false'], 'Set to true to remove any whitespace characters preceding newline characters and false to ensure it doesn\'t.')
+	];
+
+	// =========================================================================
+	// PUBLIC INTERFACE
+	// =========================================================================
+	public provideCompletionItems(document: TextDocument, position: Position, token: CancellationToken): CompletionItem[] {
+		// get text where code completion was activated
+		let rangeFromLineStart = new Range(new Position(position.line, 0), position);
```
Comment (SamVerschueren) [inline src/EditorConfigCompletionProvider.ts]: It's nitpicking, I know, but all the `let`'s could be replaced with `const`.

Comment (jedmao) [inline src/EditorConfigCompletionProvider.ts]: We can't ask him to do that until we introduce [this guy](https://palantir.github.io/tslint/rules/prefer-const/) into the project.

Comment (SamVerschueren) [inline src/EditorConfigCompletionProvider.ts]: Good point!

**Comment by jedmao** [comment]
Just waiting for you to add `unset` and I'll squash/merge.

**Comment by jedmao** [comment]
Oh wait, it's already there! 😆

**Comment by SamVerschueren** [comment]
Great work @dinhani! 🍻

**Comment by dinhani** [comment]
@jedmao @SamVerschueren 

There are two things I still want to work on this topic and your feedback would be helpful.

1) Should the "equals (=)" sign be automatically added after the user select a property name? 

I think it is possible using CompletionItem#additionalTextEdits field
https://code.visualstudio.com/Docs/extensionAPI/vscode-api#_a-namecompletionitemaspan-classcodeitem-id762completionitemspan)

2) Values are displayed sorted alphabetically. So it displays "false, true, unset". And sometimes "unset" is in the middle of the other values like "latin1, unset, utf-8". 

I'd like to display the order of boolean to "true, false" and in all cases display "unset" always as the last option? It is possible to do this using the "sortText" option.

What do you think?

**Comment by jedmao** [comment]
1. I think the `=` would be helpful. They would never NOT add one.
1. I also think the order is important, but I'm not sure I know the answer of "how", so however you figure out how to do that, I support it.

**Comment by SamVerschueren** [comment]
1. I agree, `=` would be helpfull
2. I would sort them alphabetically and then add `unset` as last item in the list (if possible)

**Comment by jedmao** [comment]
I think `true`, `false` makes a lot of sense NOT to be alphabetical.

**Comment by jedmao** [comment]
Published in version `0.4.0`.

**Comment by SamVerschueren** [comment]
Awesome!!!!

**Pull request merged**